src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,120 @@
+import * as RadixPopover from "@radix-ui/react-popover";
+import { Arrow, GenericMenuPanel } from "../GenericMenu";
+import styled from "styled-components";
+import { ReactNode } from "react";
+import { Icon } from "..";
+import { EmptyButton } from "../FormField/commonElement";
+
+export const Popover = ({ children, ...props }: RadixPopover.PopoverProps) => {
+  return <RadixPopover.Root {...props}>{children}</RadixPopover.Root>;
+};
+
+const Trigger = styled(RadixPopover.Trigger)`
+  width: fit-content;
+`;
+
+interface TriggerProps extends RadixPopover.PopoverTriggerProps {
+  anchor?: ReactNode;
+}
+
+const PopoverTrigger = ({ children, anchor, ...props }: TriggerProps) => {
+  return (
+    <>
+      <Trigger
+        asChild
+        {...props}
+      >
+        {children}
+      </Trigger>
+      {anchor && <RadixPopover.Anchor asChild>{anchor}</RadixPopover.Anchor>}
+    </>
+  );
+};
+PopoverTrigger.displayName = "PopoverTrigger";
+Popover.Trigger = PopoverTrigger;
+
+interface PopoverContentProps extends RadixPopover.PopoverContentProps {
+  showArrow?: boolean;
+  showClose?: boolean;
+  forceMount?: true;
+  container?: HTMLElement | null;
+}
+
+const MenuPanel = styled(GenericMenuPanel)<{ showClose?: boolean; showArrow?: boolean }>`
+  padding: 0.5rem 1rem;
+
+  ${({ showClose }) =>
+    showClose
+      ? `
+    padding-top: 1rem;
+  `
+      : ""};
+  ${({ showArrow }) =>
+    showArrow
+      ? `
+    margin: -1px 0;
+  `
+      : ""};
+`;
+
+const CloseButton = styled(EmptyButton)`
+  position: absolute;
+  top: 0px;
+  right: 0px;
+`;
+
+const PopoverContent = ({
+  children,
+  showArrow,
+  showClose,
+  forceMount,
+  container,
+  ...props
+}: PopoverContentProps) => {
+  return (
+    <RadixPopover.Portal
+      forceMount={forceMount}
+      container={container}
+    >
+      <MenuPanel
+        as={RadixPopover.Content}
+        type="popover"
+        showClose={showClose}
+        showArrow={showArrow}
+        {...props}
+      >
+        {showClose && (
+          <CloseButton
+            as={RadixPopover.Close}
+            asChild
+          >
+            <Icon name="cross" />
+          </CloseButton>
+        )}
+        {showArrow && (
+          <Arrow
+            asChild
+            as={RadixPopover.Arrow}
+          >
+            <svg
```

Comment:
we were creating an icon component and now we are using svg directly here? I don't think it is a good idea

Reply:
I added this as a dummy value and this would only be used inside the arrow option and not outside in our apps
So I added directly
Will move it to a separate component
But don't think it should recide inside Icon

Reply:
why not?